lib/KalturaFTPServer.js

```diff
@@ -241,11 +254,35 @@ FtpSession.prototype = {
 		this.lists[service] = [];
 
 		var currentList = [];
-		var filter = this.createFilter(map);
+		var filter = this.createListingObject(map, 'filter', {});
+		var pager = this.createListingObject(map, 'pager', null);
 
-		var pager = this.createPager(map);
+		var retrieveOnePageOnly = true;
+
+		if ( pager == null ){
+			pager = new kaltura.client.objects.KalturaFilterPager();
+			pager.pageIndex = 1;
+			pager.pageSize = 500;
+			var retrieveOnePageOnly = false;
+		}
 
 		var This = this;
+
+		var pagingHandler = function (err, currentList, count) {
```

Comment:
i need to compare to pager.pageSize in order to continue calling listAction. currentList can be the pageSize only on first iteration. i need to count how many items where retrieved since i don't have the total count i all formats.